src/lib/Libutil/misc_utils.c

```diff
@@ -2259,3 +2260,67 @@ get_msvr_mode(void)
 {
 	return 0;
 }
+
+/**
+ * @brief
+ * 		state_char2int - return the state from character form to int form.
+ *
+ * @param[in]	stc	-	state in character form
+ *
+ * @return	state in int form
+ * @retval	-1	: failure
+ */
+
+int
+state_char2int(char stc)
+{
+	int  i;
+	char statechars[] = "TQHWREXBMF";
```

Comment:
Don't we need to include 'S' and 'U' here?

Reply:
No, the numeric states have historically not considered U and S, they are included under R. I thought about adding U and S to make them first class citizens in server code but it is not trivial, we can do that refactoring later.